crates/common/logged_command/src/logged_command.rs

```diff
@@ -16,18 +18,60 @@ pub struct LoggingChild {
 
 impl LoggingChild {
     pub async fn wait_with_output(
-        self,
+        mut self,
         logger: &mut BufWriter<File>,
+        time_out: Option<usize>,
     ) -> Result<Output, std::io::Error> {
+        match time_out {
+            Some(timeout) => {
+                tokio::select! {
+                    _ = self.inner_child.wait() => {
+                        self.get_outcome(logger).await
+                    }
+                    _ = tokio::time::sleep(std::time::Duration::from_secs(timeout.try_into().unwrap())) => {
+                        self.timeout_operation(logger, timeout).await
+                    }
```

Comment:
This is not clear and hides the same issue as the first arm of the `tokio::select`, i.e. calling different `wait` then `wait_with_output` making unclear which part of the output is collected.

Reply:
not valid anymore